File: CS/Models/Product.cs
using System.Web.Mvc;
using System.ComponentModel.DataAnnotations;

public class Product {
    public int ProductID { get; set; }
    [Required(ErrorMessage = "Product Name is required")]
    [Remote("CheckIfExists", "Home", AdditionalFields = "OriginalProductName", ErrorMessage = "The name you specified already exists. Please specify a different name.")]
    public string ProductName { get; set; }
    public decimal? UnitPrice { get; set; }
    public short? UnitsOnOrder { get; set; }
}

File: VB/Models/Product.vb
Imports System.Web.Mvc
Imports System.ComponentModel.DataAnnotations

Public Class Product
	Public Property ProductID() As Integer
	<Required(ErrorMessage := "Product Name is required"), Remote("CheckIfExists", "Home", AdditionalFields := "OriginalProductName", ErrorMessage := "The name you specified already exists. Please specify a different name.")>
	Public Property ProductName() As String
	Public Property UnitPrice() As Decimal?
	Public Property UnitsOnOrder() As Short?
End Class